### ISSUE TITLE
Cannot rasterize the layer in worker thread

### ISSUE THREAD

**Issue by ezioda004** [paperjs/paper.js#1643]
I've bunch of imported projects and would like to rasterize them to create images in the worker thread but I'm unable to do that,
The error I get is:
```
Uncaught TypeError: Cannot read property 'getContext' of null at Layer.rasterize
```
I guess its because of canvas/DOM being unavailable in the worker but I've even tried using `OffscreenCanvas` and still the same result.
So is it possible to rasterize in the worker thread?

Any help would be appreciated.
Thanks.

**Comment by sapics** [comment]
Hello @ezioda004 !

I guess that [paper-full.zip](https://github.com/paperjs/paper.js/files/2955746/paper-full.zip) would work in worker?

**Comment by lehni** [comment]
@sapics could you explain what you changed? @ezioda004 does this work for you?

**Comment by sapics** [comment]
Yes, I changed as follows.
```diff
- canvas = document.createElement('canvas');
+ canvas = new OffscreenCanvas();
```

**Comment by MuTsunTsai** [comment]
I don't think that alone will work, as there're too many lines of code that target only DOM environment. Correct me if I'm wrong, but my understanding is that, for the moment, paper.js simply doesn't support `OffscreenCanvas`. Its current support for web worker is limited to processing vector graphics, and you need to export the result by `exportJSON()` method and import the data back in the DOM thread.
I do think that since now we have `OffscreenCanvas` at least for Chromium based browsers, it worth serious consideration that paper.js should add support to it, which will be a major performance boost.
